python/dgl/transform.py

```diff
@@ -945,8 +946,11 @@ def remove_edges(g, edge_ids):
                 "Graph has more than one edge type; specify a dict for edge_id instead.")
         edge_ids = {g.canonical_etypes[0]: edge_ids}
 
-    edge_ids_nd = [nd.null()] * len(g.etypes)
+    edge_ids_nd = [nd.null(g._graph.dtype)] * len(g.etypes)
     for key, value in edge_ids.items():
+        if value.dtype != getattr(F, g._graph.dtype):
+            # if didn't check, this function still works, but returns wrong result
+            raise Exception("Inconsistent dtype between edge ids and graph data type")
```

Comment:
I would like to see a helper function like this:
```python
check_idtype_dict({k : v.idtype for k,v in edge_ids}, g.idtype)
```
You could also have things like
```python
check_idtype_list(idtype_list, g.idtype)
```